graphic_templates/waffle_chart/js/graphic.js

```diff
@@ -0,0 +1,293 @@
+        .enter()
+        .call( function(selection) {
+            // Render bottom layer of squares
+            selection.append('rect')
+                .attr("width", squareSize - gap)
+                .attr("height", squareSize - gap)
+                .attr("fill", function(d) {
+                    return colorScale(d.groupIndex);
+                })
+                .attr("x", function(d, i)
+                {
+                    col = i%widthSquares;
+                    var x = (col * (squareSize - gap)) + (col * gap); 
+                    return x;
+                })
+                .attr("y", function(d, i) {
+                    //group n squares for column
+                    row = Math.floor(i/widthSquares);
+                    return (row * (squareSize - gap)) + (row*gap);
+                })
+
+            /* Render top layer of squares
+            ----------------------------------------------------*/
+            selection.append('rect')
+                .attr("width", squareSize - gap)
+                .attr("height", squareSize - gap)
+                .attr("fill", function (d,i) {
+                     switch (d.groupIndex % 4) {
+                        case 0: return "rgba(255,255,255,0.0)"; // fully transparent
+                        case 1: return "url(#diagonalHatchLeft)";
+                        case 2: return "rgba(255,255,255,0.0)"; // fully transparent
+                        case 3: return "url(#diagonalHatchRight)";
+                    }
+                })
+                .attr("x", function(d, i) {
+                    col = i%widthSquares;
+                    var x = (col * (squareSize - gap)) + (col * gap); 
+                    return x;
+                })
+                .attr("y", function(d, i) {
+                    //group n squares for column
+                    row = Math.floor(i/widthSquares);
+                    return (row * (squareSize - gap)) + (row*gap);
+                })
+                .on("mouseover", function(d, i) {
+                    return tooltip.style("visibility", "visible");
+                })
+                .on("click", function(d, i) { // for mobile
+                    return tooltip.style("visibility", "visible");
+                })
+                .on("mousemove", function(d, i) {
+                    tooltip.html('<div><strong>' + DATA[d.groupIndex].label + "</strong></div><div>"
+                    +  d.amt + ", " + d.units + "%</div>");
+                    return tooltip.style("top", (d3.event.pageY - 10)+"px").style("left",(d3.event.pageX + 14)+"px");
+                })
+                .on("mouseout", function(d, i) {
+                    return tooltip.style("visibility", "hidden");
+                });
+        });
+
+
+    /*
+     * Output the legend
+     */
+
+    // Create a legend div wrapper
+    var chartLegend = chartWrapper.append("ul")
+        .attr('class', 'labels')
+        .style({
+            width: labelWidth + 'px',
```

Comment:
I'd prefer to see any style declarations that aren't dynamic and based on JS variables declared in the stylesheet rather than in the javascript.